x-pack/plugins/endpoint/server/services/resolver/query_builder.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { isLegacyEntityID, parseLegacyEntityID, CountQueryInfo } from './common';
+import { EndpointAppContext, JSONish } from '../../types';
+import { EndpointAppConstants } from '../../../common/types';
+
+export interface PaginationInfo {
+  page: number | undefined;
+  pageSize: number | undefined;
+}
+
+function buildLegacyChildrenQuery(endpointID: string, uniquePID: string) {
+  return {
+    bool: {
+      filter: {
+        bool: {
+          should: [
+            {
+              bool: {
+                filter: [
+                  {
+                    term: { 'endgame.unique_pid': uniquePID },
+                  },
+                  {
+                    // had to change these to match queries otherwise I can't get results back, using the kibana
+                    // developer mode it works fine AND the api tests when it's also a term...not sure why
+                    match: { 'agent.id': endpointID },
+                  },
+                ],
+              },
+            },
+            {
+              bool: {
+                filter: [
+                  {
+                    term: { 'endgame.unique_ppid': uniquePID },
+                  },
+                  {
+                    match: { 'agent.id': endpointID },
+                  },
+                ],
+              },
+            },
+          ],
+        },
+      },
+    },
+  };
+}
+
+function buildESEndpointChildrenQuery(entityID: string) {
+  return {
+    bool: {
+      filter: {
+        bool: {
+          should: [
+            {
+              match: { 'endpoint.process.entity_id': entityID },
+            },
+            {
+              match: { 'endpoint.process.parent.entity_id': entityID },
+            },
+          ],
+        },
+      },
+    },
+  };
+}
+
+export async function getESChildrenQuery(
+  context: EndpointAppContext,
+  entityID: string,
+  paginationInfo: PaginationInfo
+) {
+  const { index, query } = getESChildrenCountQuery(entityID);
+  return await buildSearchBody(context, query, paginationInfo, index);
+}
+
+export function getESChildrenCountQuery(entityID: string): CountQueryInfo {
```

Comment:
If I'm reading this correctly this function and `getESNodeCountQuery` have misleading names since they're returning a query which is used to get both the full document hits and the document counts - the query is not specific to the count API.

Reply:
Yeah, I'll refactor it.